src/docfx/docset/Docset.cs

```diff
@@ -85,30 +90,33 @@ internal class Docset
 
         private readonly CommandLineOptions _options;
         private readonly Report _report;
+        private readonly ConcurrentDictionary<string, Repository> _repositories;
```

Comment:
Should this be `ConcurrentDictionary<string, Lazy<Repository>>`? so that the value will be executed just once, and no need to have `Lazy` in `Document.cs` while read from it

Reply:
u r right!